lib/jsmidgen.js

```diff
@@ -393,8 +393,15 @@ var Midi = {};
 		var trackCount = this.tracks.length.toString(16);
 
 		// prepare the file header
-		var bytes = File.HDR_CHUNKID + File.HDR_CHUNK_SIZE + File.HDR_TYPE0;
+		var bytes = File.HDR_CHUNKID + File.HDR_CHUNK_SIZE;
 
+		// set Midi type based on number of tracks
+		if (parseInt(trackCount) > 1) {
```

Comment:
Can there be more than 10 tracks? If so, then this code will fail. `trackCount` is a `hex` value. In case of 10, it s value will be `a`. Maybe you can fix that by adding `16` as the `radix` of the `parseInt` function. 

Other than that what about Type 2 files, where each track represents an independent sequence? (I believe Type0 is where it has a single track and Type1 is the one which contain one or more simultaneous tracks). I think if we are extending support to the formats, then we should do it for all 3 types. Though I ll let @dingram take that call.